auth/hook.go

```diff
@@ -300,3 +304,17 @@ func (hf *hookFields) GetArray(k string, dv []string) []string {
 	}
 	return dv
 }
+
+func (hf *hookFields) GetRules(k string, dv []rules.Rule) []rules.Rule {
+	val, ok := hf.Values[k]
+	if !ok {
+		return dv
+	}
+	var Rules []rules.Rule
```

Comment:
In go by convention all the variables should start from the lower case. You can rename it to `userRules` or something like that